queue/gc_queue.go

```diff
@@ -15,6 +15,22 @@ type Recycler interface {
 	Recycle()
 }
 
+// RecycleSlice zeroes byte slice, making it fresh for another use.
+// Zeroing the buffer using a logarithmic number of calls to the optimized copy
+// method.  Benchmarking shows this to be ~30 times faster than a for loop that
+// sets each index to 0 for this buffer size. Inspired by:
+// https://stackoverflow.com/questions/30614165/is-there-analog-of-memset-in-go
+func RecycleSlice(b []byte) {
```

Comment:
Don't see how this fits within the `queue` package, but fine with me.

Reply:
yes that bothered me too, i'm thinking it more appropriately belongs in a `buffer` package. i can add this lone method to that package now, which would unblock #2602. does that sound good?

Reply:
Yep!

Reply:
kk will fix